diff --git a/package/qq/package.nix b/package/qq/package.nix
index f5933b5..d217a4e 100644
--- a/package/qq/package.nix
+++ b/package/qq/package.nix
@@ -25,7 +25,7 @@
   at-spi2-core,
   autoPatchelfHook,
   makeShellWrapper,
-  wrapGAppsHook3,
+  wrapGAppsHook4,
   commandLineArgs ? "",
 }:
 
@@ -48,7 +48,7 @@ let
       "x86_64-darwin"
       "x86_64-linux"
     ];
-    license = lib.licenses.unfree;
+    # license = lib.licenses.unfree;
     sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
     maintainers = with lib.maintainers; [
       bot-wxt1221
@@ -93,7 +93,7 @@ else
     nativeBuildInputs = [
       autoPatchelfHook
       makeShellWrapper
-      wrapGAppsHook3
+      wrapGAppsHook4
       dpkg
     ];
 
@@ -127,9 +127,9 @@ else
       cp -r opt $out/opt
       cp -r usr/share $out/share
       substituteInPlace $out/share/applications/qq.desktop \
-        --replace-fail "/opt/QQ/qq" "$out/bin/qq" \
+        --replace-fail "/opt/QQ/qq" "$out/opt/QQ/qq" \
         --replace-fail "/usr/share" "$out/share"
-      makeShellWrapper $out/opt/QQ/qq $out/bin/qq \
+      makeShellWrapper $out/opt/QQ/qq $out/opt/QQ/qq-wrapper \
         --prefix XDG_DATA_DIRS : "$GSETTINGS_SCHEMAS_PATH" \
         --prefix LD_PRELOAD : "${lib.makeLibraryPath [ libssh2 ]}/libssh2.so.1" \
         --prefix LD_LIBRARY_PATH : "${
